lib/mirage.ml

```diff
@@ -1762,20 +1783,23 @@ let ldflags pkg = pkg_config pkg ["--variable=ldflags"]
 
 let ldpostflags pkg = pkg_config pkg ["--variable=ldpostflags"]
 
-let link info name target =
+let link info target =
   let libs = Info.libraries info in
+  let output = output info in
+  Bos.OS.Cmd.run Bos.Cmd.(v "mkdir" % "-p" % Filename.dirname output)
```

Comment:
`Bos.OS.Dir.create` (see http://erratique.ch/software/bos/doc/Bos.OS.Dir.html#VALcreate) will work on more platforms and do the same AFAICS